sensors/sqs_sensor.py

```diff
@@ -98,50 +111,117 @@ def _may_setup_sqs(self):
-                self.session.region_name == self.aws_region
-
-        if self.session is None or not _is_same_credentials():
-            self._setup_sqs()
-
-    def _setup_sqs(self):
-        ''' Setup Boto3 structures '''
-        self._logger.debug('Setting up SQS resources')
-        self.session = Session(aws_access_key_id=self.aws_access_key,
-                               aws_secret_access_key=self.aws_secret_key,
-                               region_name=self.aws_region)
+        for account_id in required_accounts:
+            if account_id != self.account_id and account_id not in cross_roles_arns:
+                continue
+
+            session = self.sessions.get(account_id, None)
+            same_credentials = _is_same_credentials(session, account_id) if session else False
+
+            if session is None or not same_credentials:
+                if account_id == self.account_id:
+                    self._setup_session()
+                else:
+                    self._setup_multiaccount_session(account_id, cross_roles_arns)
 
+    def _setup_session(self):
+        ''' Setup Boto3 session '''
+        session = Session(aws_access_key_id=self.access_key_id,
+                          aws_secret_access_key=self.secret_access_key)
+
+        if not self.account_id:
+            self.account_id = session.client('sts').get_caller_identity().get('Account')
+            self.credentials[self.account_id] = (self.access_key_id, self.secret_access_key, None)
+
+        self.sessions[self.account_id] = session
+
+    def _setup_multiaccount_session(self, account_id, cross_roles_arns):
+        ''' Assume role and setup session for the cross-account capability'''
         try:
-            self.sqs_res = self.session.resource('sqs')
-        except NoRegionError:
-            self._logger.warning("The specified region '%s' is invalid", self.aws_region)
+            assumed_role = self.sessions[self.account_id].client('sts').assume_role(
+                RoleArn=cross_roles_arns[account_id],
+                RoleSessionName='StackStormEvents'
+            )
+        except ClientError:
+            self._logger.error('Could not assume role on %s', account_id)
+            return
+
+        self.credentials[account_id] = (assumed_role["Credentials"]["AccessKeyId"],
+                                        assumed_role["Credentials"]["SecretAccessKey"],
+                                        assumed_role["Credentials"]["SessionToken"])
+
+        session = Session(
+            aws_access_key_id=self.credentials[account_id][0],
+            aws_secret_access_key=self.credentials[account_id][1],
+            aws_session_token=self.credentials[account_id][2]
+        )
+        self.sessions[account_id] = session
+
+    def _setup_sqs(self, session, account_id, region):
+        ''' Setup SQS resources'''
+        try:
+            if not self.sqs_res.get(account_id, None):
+                self.sqs_res[account_id] = {}
```

Comment:
Use collections defaultdict `self.sqs_res = defaultdict(dict)`, then you can get rid of this if